sanic_routing/patterns.py

```diff
@@ -10,10 +10,10 @@ def parse_date(d):
 REGEX_PARAM_NAME = re.compile(r"^<([a-zA-Z_][a-zA-Z0-9_]*)(?::(.*))?>$")
 REGEX_TYPES = {
     "string": (str, re.compile(r"^[^/]+")),
-    "int": (int, re.compile(r"^-?\d+")),
-    "number": (float, re.compile(r"^-?(?:\d+(?:\.\d*)?|\.\d+)")),
     "alpha": (str, re.compile(r"^[A-Za-z]+")),
     "path": (str, re.compile(r"^[^/]?.*?")),
+    "number": (float, re.compile(r"^-?(?:\d+(?:\.\d*)?|\.\d+)")),
+    "int": (int, re.compile(r"^-?\d+")),
```

Comment:
Not important, but curious why these were reversed and reordered?

Reply:
When you had multiple routes with dynamic parts at the same spot, it was sorting based upon an alphabetic: `<foo:string>, <foo:int>, <foo:uuid>`. The order of these was somewhat arbitrary.

```python
    def _sorting(item) -> t.Tuple[bool, int, str, int]:
        key, child = item
        type_ = 0
        if child.dynamic:
            key = key[1:-1]
            if ":" in key:
                key, param_type = key.split(":")
                try:
                    type_ = list(REGEX_TYPES.keys()).index(param_type)
                except ValueError:
                    type_ = len(list(REGEX_TYPES.keys()))
        return child.dynamic, len(child._children) * -1, key, type_ * -1
```

Now, we use only the param name, and then add an additional sort check that is based upon the order defined in patterns.py.

Reply:
Ok, so now tree generation will be more reproducible/predictable.